plugins/modules/s3_bucket.py

```diff
@@ -691,7 +763,8 @@ def main():
         versioning=dict(type='bool'),
         ceph=dict(default=False, type='bool'),
         encryption=dict(choices=['none', 'AES256', 'aws:kms']),
-        encryption_key_id=dict()
+        encryption_key_id=dict(),
+        public_access=dict(type='dict')
```

Comment:
```suggestion
        public_access=dict(type='dict', options=dict(
            block_public_acls=dict(type='bool'),
            ignore_public_acls=dict(type='bool'),
            block_public_policy=dict(type='bool'),
            restrict_public_buckets=dict(type='bool')),
```

This can be converted to the format used by boto3 using snake_dict_to_camel_dict (and back again using camel_dict_to_snake_dict) these functions can be found in ansible.module_utils.common.dict_transformations

Reply:
Sure, that will be much better. I was not aware about this 'dict in dict'.